src/widgets/dialogs/EditHotkeyDialog.cpp

```diff
@@ -0,0 +1,296 @@
+#include "widgets/dialogs/EditHotkeyDialog.hpp"
+
+#include "Application.hpp"
+#include "common/QLogging.hpp"
+#include "controllers/hotkeys/ActionNames.hpp"
+#include "controllers/hotkeys/HotkeyController.hpp"
+#include "controllers/hotkeys/HotkeyHelpers.hpp"
+#include "ui_EditHotkeyDialog.h"
+
+namespace chatterino {
+
+EditHotkeyDialog::EditHotkeyDialog(const std::shared_ptr<Hotkey> hotkey,
+                                   bool isAdd, QWidget *parent)
+    : QDialog(parent, Qt::WindowStaysOnTopHint)
+    , ui_(new Ui::EditHotkeyDialog)
+    , data_(hotkey)
+{
+    this->ui_->setupUi(this);
+    const auto app = getApp();
+    // dynamically add category names to the category picker
+    for (const auto &[_, hotkeyCategory] : app->hotkeys->categories())
+    {
+        this->ui_->categoryPicker->addItem(hotkeyCategory.displayName,
+                                           hotkeyCategory.name);
+    }
+
+    this->ui_->warningLabel->hide();
+
+    if (hotkey)
+    {
+        if (!hotkey->validAction())
+        {
+            this->showEditError("Invalid action, make sure you select the "
+                                "correct action before saving.");
+        }
+
+        // editing a hotkey
+
+        // update pickers/input boxes to values from Hotkey object
+        this->ui_->categoryPicker->setCurrentIndex(size_t(hotkey->category()));
+        this->ui_->keyComboEdit->setKeySequence(
+            QKeySequence::fromString(hotkey->keySequence().toString()));
+        this->ui_->nameEdit->setText(hotkey->name());
+        // update arguments
+        bool isFirst = true;
+        QString argsText;
+        for (const auto arg : hotkey->arguments())
+        {
+            if (!isFirst)
+            {
+                argsText += '\n';
+            }
+            argsText += arg;
+            isFirst = false;
+        }
```

Comment:
This solution with `isFirst` could be improved by using a regular for loop with `int i = 0` and just check `if (i == 0)` I think. Besides, loop variable `arg` creates a copy, so this can be for sure improved.

Reply:
```suggestion
        QString argsText;
        for (int i = 0; i < hotkey->arguments(); I++)
        {
            if (I != 0)
            {
                argsText += '\n';
            }
            const auto &arg = hotkey->arguments().at(0);
            argsText += arg;
        }
```
Mobile coding here we fucking go